plugins/woocommerce/src/Internal/EmailEditor/Integration.php

```diff
@@ -200,4 +204,82 @@ public function extend_template_post_api(): void {
 			)
 		);
 	}
+
+	/**
+	 * Filter email preview data to replace placeholders with actual content.
+	 *
+	 * This method retrieves the appropriate email type based on the request,
+	 * generates the email content using the WooContentProcessor, and replaces
+	 * the placeholder in the preview HTML.
+	 *
+	 * @param string $data       The preview data.
+	 * @param string $email_type The email type identifier (e.g., 'customer_processing_order').
+	 * @return string The updated preview data with placeholders replaced.
+	 */
+	private function update_email_preview_data( $data, string $email_type ) {
+		$default_type_param = 'WC_Email_Customer_Processing_Order';
+		$type_param         = $default_type_param;
+
+		if ( ! empty( $email_type ) ) {
+			$type_param = 'WC_Email_' . implode( '_', array_map( 'ucfirst', explode( '_', $email_type ) ) );
+		}
+
+		$email_preview = wc_get_container()->get( EmailPreview::class );
+
+		try {
+			$message = $email_preview->generate_placeholder_content( $type_param );
+		} catch ( \InvalidArgumentException $e ) {
+			// If the provided type was invalid, fall back to the default.
+			try {
+				$message = $email_preview->generate_placeholder_content( $default_type_param );
+			} catch ( \Throwable $e ) {
+				return $data;
+			}
+		} catch ( \Throwable $e ) {
+			return $data;
+		}
+
+		return str_replace( BlockEmailRenderer::WOO_EMAIL_CONTENT_PLACEHOLDER, $message, $data );
+	}
+
+	/**
+	 * Filter email preview data used when sending a preview email.
+	 *
+	 * @param string $data The preview data.
+	 * @return string The updated preview data with placeholders replaced.
+	 */
+	public function update_send_preview_email_rendered_data( $data ) {
+		$email_type = '';
+		$post_body  = file_get_contents( 'php://input' );
+
+		if ( $post_body ) {
+			$decoded_body = json_decode( $post_body );
+
+			if ( json_last_error() === JSON_ERROR_NONE && isset( $decoded_body->postId ) ) { // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				$post_id = absint( $decoded_body->postId ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				$post    = get_post( $post_id );
+
+				if ( $post instanceof \WP_Post ) {
+					$email_type = $post->post_name;
+					return $this->update_email_preview_data( $data, $email_type );
```

Comment:
Since we have the `$post` object here, please get the WC email type using post_meta
e.g
```php
$email_type = get_post_meta( $post_id, self::WC_EMAIL_TYPE_ID_POST_META_KEY, true );
```

Here is an example https://github.com/woocommerce/woocommerce/blob/1b24ababb98cea1a215f059aacf48238fca3f33d/plugins/woocommerce/src/Internal/EmailEditor/Integration.php#L184

We want to stop associating WC email type with the email editor content post title.